### PR TITLE
Bump minimum python version to 3.7

### PR DESCRIPTION
**Pull request by Empact**
Python 3.6 is set to reach end-of-life in 2021-12-23.
https://www.python.org/dev/peps/pep-0494/#lifespan
https://docs.python.org/3/whatsnew/3.7.html

### CODE REVIEW

**Comment by Empact** [comment]
Previously updated in https://github.com/bitcoin/bitcoin/pull/19504
Python 3.7 EOL is set for 2023-06-27
https://www.python.org/downloads/

**Review by maflcko — CHANGES_REQUESTED** [review]
Doesn't work in the current state

File: ci/test/00_setup_env_native_nowallet.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 export LC_ALL=C.UTF-8
 
 export CONTAINER_NAME=ci_native_nowallet
-export DOCKER_NAME_TAG=ubuntu:18.04  # Use bionic to have one config run the tests in python3.6, see doc/dependencies.md
+export DOCKER_NAME_TAG=ubuntu:18.04  # Use bionic to have one config run the tests in python3.7, see doc/dependencies.md
 export PACKAGES="python3-zmq clang-7 llvm-7 libc++abi-7-dev libc++-7-dev"  # Use clang-7 to test C++17 compatibility, see doc/dependencies.md
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_nowallet.sh:11]: `python3-zmq` installs `python3` (3.6), which is picked by the functional tests (shebang `python3`), so this doesn't actually work

**Review by maflcko — COMMENTED** [review]
This doesn't work, but https://github.com/bitcoin/bitcoin/pull/26226 should work

**Pull request closed without merging**